src/extensions/agent_modules/device_update_workflow/module/src/device_update_workflow_module.c

```diff
@@ -104,35 +107,42 @@ int DeviceUpdateWorkflowModule_DoWork(ADUC_AGENT_MODULE_HANDLE handle)
  */
 ADUC_AGENT_MODULE_HANDLE DeviceUpdateWorkflowModule_Create()
 {
-    bool ret = false;
-    ADUC_WorkflowData *workflowData =  NULL;
     ADUC_AGENT_MODULE_HANDLE handle = NULL;
-    ADUC_AGENT_MODULE_INTERFACE* interface = NULL;
 
-    ADUC_ALLOC(interface);
+    ADUC_AGENT_MODULE_INTERFACE* tmp = NULL;
+    ADUC_WorkflowData *workflowData =  NULL;
+
+    tmp = calloc(1, sizeof(*tmp));
+    if (tmp == NULL)
+    {
+        goto done;
+    }
 
     // TODO (nox-msft) - pass in launchArgs
-    ret = AzureDeviceUpdateCoreInterface_Create((void**)&workflowData, 0, NULL);
-    if (!ret)
+    if (!AzureDeviceUpdateCoreInterface_Create((void**)&workflowData, 0, NULL))
```

Comment:
why the void** cast? Does this method not take a ADUC_WorkflowData** ?

Reply:
No, strangely it takes a formal arg of type (void**) and does not even use the context arg--no idea why!  Removing it...